### DISCUSSION TITLE
Is there a way to limit the time the mic stays open?

### DISCUSSION THREAD

**Discussion by luisffee** [KoljaB/RealtimeSTT#142] [Q&A] [answered]
I am using the library in a situation where there's a lot of noise around, with a somewhat predictable timeframe which the stt should capture the audio( A specific person makes questions in sequence ). What happens is that sometimes, when there are other people nearby speaking at the same time i start recording, the mic stays open for too long, increasing the processing time of the model and also poluting the audio which i extract the question the person in relevance makes.

**Comment by KoljaB** [comment]
This is mostly due to silero vad detecting "speech" where there is only background noise. First thing to do would be to reduce silero vad sensitivity and to set silero_deactivity_detection to True. Depending on the noise level you would need a more sophisticated approach:

What gives more stable results but is also more complicated to implement would be to check the realtime transcription for changes. If there is no additional text incoming for a while we consider speech is now finished and end recording. You can find an example implementation [here](https://github.com/KoljaB/RealtimeSTT/blob/master/tests/realtimestt_speechendpoint.py#L192-L218).

**Reply by luisffee** [reply]
Thanks for the answer, I will look further into it. Another thought perhaps, would using the abort method be a possible solution? Or would the text received so far be lost?

**Reply by KoljaB** [reply, marked as answer]
If you call abort() the text() method will return an empty string (""). 

But I'm not sure if stop() works 100% correctly here. The stop() method was intended to stop a recording which was started with the start() method before. So for manual recording. We use it here for recording that was started by VAD, using the text() method.

The abort() method was intended to quick end such a VAD initiated recording without causing a final transcription. So it's not really the right approach here, but tbh the stop method isn't either, because it will initiate the transcription but it leaves the text method in an undefined state. I need to think about that, maybe calling abort() after stop() method helps to make it reliable because this forces the text() method to leave in a defined state. But not sure about side effects, I need to test that scenario more.

**Reply by homelab-00** [reply]
You could take a look at [`realtimestt_test_hotkeys_v2.py`](https://github.com/KoljaB/RealtimeSTT/blob/master/tests/realtimestt_test_hotkeys_v2.py) in the tests folder. It has mute/unmute functionality.

edit: Just remember to change line 167 from:
`'realtime_model_type': 'Systran/faster-distil-whisper-large-v3',  # Using the same model for realtime`
to:
`'realtime_model_type': 'tiny.en',  # Or other small model`

**Reply by luisffee** [reply]
Thanks @homelab-00  and @KoljaB for the suggestions.
### So, an update to this:
I have not thoroughly tested as @KoljaB mentioned, however so far it seems the stop() method has been successful in stopping the VAD initiated recording, which is enough for me as I needed only a hard stop without losing the recorded audio so far.
